curvefs/src/metaserver/s3compact_wq_impl.cpp

```diff
@@ -295,22 +295,24 @@ int S3CompactWorkQueueImpl::ReadFullChunk(
     return 0;
 }
 
-MetaStatusCode S3CompactWorkQueueImpl::UpdateInode(CopysetNode* copysetNode,
-                                                   const PartitionInfo& pinfo,
-                                                   const Inode& inode) {
-    UpdateInodeRequest request;
+MetaStatusCode S3CompactWorkQueueImpl::UpdateInode(
+    CopysetNode* copysetNode, const PartitionInfo& pinfo, uint64_t inodeId,
+    ::google::protobuf::Map<uint64_t, S3ChunkInfoList>&& s3ChunkInfoAdd,
+    ::google::protobuf::Map<uint64_t, S3ChunkInfoList>&& s3ChunkInfoRemove) {
+    AppendS3ChunkInfoRequest request;
     request.set_poolid(pinfo.poolid());
     request.set_copysetid(pinfo.copysetid());
     request.set_partitionid(pinfo.partitionid());
     request.set_fsid(pinfo.fsid());
-    request.set_inodeid(inode.inodeid());
-    *request.mutable_s3chunkinfomap() = inode.s3chunkinfomap();
-    UpdateInodeResponse response;
-    S3CompactWorkQueueImpl::UpdateInodeClosure done;
+    request.set_inodeid(inodeId);
+    *request.mutable_s3chunkinfoadd() = std::move(s3ChunkInfoAdd);
```

Comment:
::google::protobuf::Map<uint64_t, S3ChunkInfoList>&& s3ChunkInfoAdd is rvalue, why here use std::move to convert again?

Reply:
Every named object is Lvalue
https://en.cppreference.com/w/cpp/language/value_category
"Even if the variable's type is rvalue reference, the expression consisting of its name is an lvalue expression; "

Reply:
why just use :google::protobuf::Map<uint64_t, S3ChunkInfoList>& s3ChunkInfoAdd?

Reply:
google c++ style guide? const& or pointer*